docs/pipeline-howto.md

```diff
@@ -1,266 +1,134 @@
-     * Github Client - GitHub OAuth `Client Id`
-     * Github Secret - Github OAuth `Client Secret`
-
-      <a href="images/howto/ARMCICD.png"><img src="images/howto/ARMCICD.png" height="90"></a>
-
-  * **Azure Credentials and Information**
-     * Azure Client Id - see how to get Azure Client Id above
-     * Azure Client Secret - see how to get Azure Client Secret above
-     * Azure Subscription Id - your Azure Subscription Id
-     * Azure Tenant Id - see how to get Azure Tenant Id above
-
-     <a href="images/howto/images/howto/ARMAzureCreds.png"><img src="images/howto/ARMAzureCreds.png" height="130"></a>
-
-  * Finish the wizard to create a `Control Plane` instance.
-  * Open the `Resource Group` that was specified for the deployment
-
-      <a href="images/howto/CPRGroup.png"><img src="images/howto/CPRGroup.png" height="90"></a>
-
-  * Take note of the PublicIP of the deployed `Control Plane`.
-
-    <a href="images/howto/AzureCPPubIP.png"><img src="images/howto/AzureCPPubIP.png" height="200"></a>
+The steps of the workflow executed by the CI/CD flow are described in the  `.pipeline.yml` file that must be placed under the root directory of the source code of the Spark application. The file has to be pushed into the GitHub repo along with the source files of the application.
 
-### Define `.pipeline.yml` pipeline workflow configuration for your Spark application
+There is an example Spark application [spark-pi-example](https://github.com/banzaicloud/spark-pi-example/tree/0.3.0) that can be used for trying out the CI/CD pipeline.
 
-The steps of the workflow executed by the CI/CD flow are described in the  `.pipeline.yml` file that must be placed under the root directory of the source code of the Spark application. The file has to be pushed into the GitHub repo along with the source files of the application.
+>Note: Fork this repository into your own repository for this purpose!.
 
-There is an example Spark application [spark-pi-example](https://github.com/banzaicloud/spark-pi-example/tree/0.2.0) that can be used for trying out the CI/CD pipeline.
+For setting up your own spark application for the workflow you can start from one of the `.pipeline.yml` configuration file templates from `spark-pi-example` and customize it. Notice that there are separate templates for the different cloud providers. Pick the one that corresponds for the cloud provider you're using and create your `.pipeline.yml` from it.
 
->Note: Fork this repository into your own repository for this purpose!).
+The following sections may need to be customized:
 
-For setting up your own spark application for the workflow you can start from the `.pipeline.yml` configuration file from `spark-pi-example` and customize it.
+- the cluster where your application executed
 
-The following sections needs to be modified:
+  ```yaml
+  create_cluster:
+    ...
+    cluster_name: "[[your-cluster-name]]"
```

Comment:
The cluster name must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,38}[a-z0-9])?)' (only alphanumerics and '-' allowed, must start with a letter and end with an alphanumeric, and must be no longer than 40 characters)

Reply:
@csendesm I added this to the doc